infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackController.kt

```diff
@@ -207,6 +207,22 @@ class LocationTrackController(
         return locationTrackService.getMetadataSections(id, publishType, boundingBox)
     }
 
+    @PreAuthorize(AUTH_ALL_READ)
+    @GetMapping("/{publicationState}/by-tracknumber/{trackNumberId}")
```

Comment:
Jälleen erikoinen rest-polku. Parempia ajatuksia?
Mainitisinkin tästä Juusen reviewissä mutta ratkaistiin siis sama kysymys vähän eri tavoilla. Juusen malli oli kyllä REST-mielessä nätimpi mutta tässä on se kiva puoli että raiteiden haku pysyy raide-controllerissa.

Reply:
Joo, itsekin vähän tuskailin noiden sijoittelujen kanssa ja päädyin omaani nimenomaan sen takia että tällä tavalla endpointtien poluista tulee jotenkin nurinkurisia. Enitenhän tässä tappelee keskenään kaksi eri ideologiaa, eli valitaanko käytetty controller
* a) palautettavan datamallin mukaan (puoltaa tätä mallia) vai
* b) sisäänannettavan datan mukaan (puoltaa omaa malliani)
Aiemmin meillä ei ole tainnut oikeastaan olla tuollaisia epäsuoria hakuja, joten näiden tapausten erottelulle toisistaan ei ole ollut kauheasti tarvetta, mutta meidän polut on kyllä selkeästi tuon b-vaihtoehdon mukaisia. Tämä on kyllä kinkkinen koska itsekin haluaisin että `LocationTrack`-kama on `LocationTrack`-nimettyjen asioiden takana. Eniten tässä on ehkä siis kyse periaatepäätöksistä sen suhteen, että mikä meillä on Controllerin pääasiallinen käytännön rooli (toteuttaa ulospäin näkyvä REST-rajapinta vs. groupata saman datamallin endpointit yhteen) ja sen suhteesta samannimiseen serviceen (controller toteuttaa rajaapintansa mitä tahansa servicejä hyödyntäen vs. controllerista pyritään mahdollisuuksien mukaan käyttämään samannimistä serviceä)

Service-tasolla tämä on mielestäni selkeämpi tilanne, siellä nämä kuuluu selkeästi `LocationTrackService`lle. Haluaisin ehkä kuitenkin edelleen kääntyä (aivan liian pitkän pohdinnan jälkeen) sille kannalle, että Controllerin ensisijainen tehtävä on vain toteuttaa ulkoapäin tarkasteltuna fiksuin mahdollinen rajapinta, koska meillä on kuitenkin sen alla service-kerros, jonka on tarkoitus hoitaa oikea bisneslogiikka. But that's just, like, my opinion, man, ja mun mielipide tämän asian osalta ei ole supervahva. Voisi ehkä jutella tästä jonkun dailyn johdannaisena ja kysellä muiltakin mielipiteitä, niin saadaan päätettyä kummalla mallilla mennään

Reply:
Olen taipuvainen olemaan samaa mieltä. Controller on olemassa vain API:n määrittelyä varten. :+1: 

Ehkä yksi lisävaihtoehto olisi että poistaisi tuon locationtrack alkuosan controller-tasolta ja laittaisi kaikkiin funkkareihin koko nimen. Sitten funktio voisi olla täällä vaikka se olisikin sisällöltään "/track-numbers/../location-tracks"